modules/internal/Atomics.chpl

```diff
@@ -278,148 +101,205 @@ module Atomics {
     chpl_rmem_consist_fence(order);
   }
 
+
+  private proc isSupported(type T) param {
+    return T == bool || isInt(T) || isUint(T) || isReal(T);
+  }
+
+  // Compute the C/Runtime type from the Chapel type
+  // TODO support extern type renaming?
+  private proc externT(type T) type {
+    extern type atomic_bool;
+
+    extern type atomic_int_least8_t;
+    extern type atomic_int_least16_t;
+    extern type atomic_int_least32_t;
+    extern type atomic_int_least64_t;
+
+    extern type atomic_uint_least8_t;
+    extern type atomic_uint_least16_t;
+    extern type atomic_uint_least32_t;
+    extern type atomic_uint_least64_t;
+
+    extern type atomic__real64;
+    extern type atomic__real32;
+
+    select T {
+      when bool     do return atomic_bool;
+
+      when int(8)   do return atomic_int_least8_t;
+      when int(16)  do return atomic_int_least16_t;
+      when int(32)  do return atomic_int_least32_t;
+      when int(64)  do return atomic_int_least64_t;
+
+      when uint(8)  do return atomic_uint_least8_t;
+      when uint(16) do return atomic_uint_least16_t;
+      when uint(32) do return atomic_uint_least32_t;
+      when uint(64) do return atomic_uint_least64_t;
+
+      when real(32) do return atomic__real32;
+      when real(64) do return atomic__real64;
+    }
+  }
+
+  private proc externTString(type T) param {
+    if T == bool then return "bool";
+    if isInt(T)  then return "int_least" + numBits(T):string + "_t";
+    if isUint(T) then return "uint_least" + numBits(T):string + "_t";
+    if isReal(T) then return "_real" + numBits(T);
```

Comment:
why do the integer versions cast `numBits(T)` to string but the real version does not? I would have thought the addition would be left-associative, and so the cast wouldn't be necessary in the int case.

Reply:
Whoops, just an omission  -- this explicit cast isn't needed but I found it to be a little bit cleaner

Reply:
Updated to add the cast